### DISCUSSION TITLE
Is uploading of a zipped file supported?

### DISCUSSION THREAD

**Discussion by PiotrNowara** [oxigraph/oxigraph#552] [Q&A] [answered]
I'm trying to upload a set of triples compressed in a GZIP file using  the /store endpoint? I'm getting 415 repsonses when trying to set Content-Type to "application/gzip" or when using multipart form (which works in Fuseki). Is this supported at all? The broader question is what kind of Content-Types are supported by this endpoint?

Thanks,
Piotr

**Comment by adamreichold** [comment]
Just as thought, wouldn't be nicer to use `Content-Encoding` to transfer compressed data and handle this transparently at the HTTP level?

**Comment by Tpt** [comment, marked as answer]
Sadly, gzip is not supported yet by the HTTP API. The supported endpoints are text/turtle, application/n-triples, application/rdf+xml, application/trig and application/n-quads. Indeed, it would be nice to support `Content-Encoding` (oxigraph/oxhttp#15).

However, gzip is supported by the `load` subcommand of `oxigraph_server`. `oxigraph_server load --location target_dir --file my_file.nt.gz` should work.

**Comment by PiotrNowara** [comment]
Hi @Tpt - thank you for a quick answer.
